pkg/upstream/cluster/loadbalancer.go

```diff
@@ -429,10 +429,27 @@ func (lb *maglevLoadBalancer) ChooseHost(ctx types.LoadBalancerContext) types.Ho
 	hash := hashPolicy.GenerateHash(ctx.DownstreamContext())
 	index := lb.maglev.Lookup(hash)
 	chosen := lb.hosts.Get(index)
+	total := lb.hosts.Size()
+
+	// if retry, means request to last chose host failed, do not use it again
+	context := ctx.DownstreamContext()
+	if ind, err := variable.GetString(context, VarProxyUpstreamIndex); err == nil {
+		if i, err := strconv.Atoi(ind); err == nil {
+			index = i + 1
+		}
+		for i := 0; i < total; i++ {
```

Comment:
maybe we could call `chooseHostFromHostList(i)`.

Reply:
we can directly go to fallback if retry.

Reply:
> we can directly go to fallback if retry.

if we use chooseHostFromHostList, consider about this situation:
```hosts{0:unheathy, 1:healthy but actually unhealthy, 2:healthy but actually unhealthy, ... }```, first time hosts[1] is chosen and then hosts[2], and then hosts[1] again, other hosts will never be chosen when retrying